src/snowflake/connector/cache.py

```diff
@@ -511,9 +532,12 @@ def _load(self) -> bool:
             logger.debug("Fail to read cache from disk due to error: %s", e)
             return False
 
-    def _save(self, load_first: bool = True) -> bool:
+    def _save(self, load_first: bool = True, force_flush: bool = False) -> bool:
         """Save cache to disk if possible, returns whether it was able to save."""
         self._clear_expired_entries()
+        if not self._cache_modified and not force_flush:
+            # cache is not updated, so there is no need to dump cache to file, we just return
+            return True
```

Comment:
It didn't save, that should be `False`
```suggestion
            return False
```

Reply:
updated